Source/Core/DolphinQt/EditTimePlayedDialog.cpp

```diff
@@ -0,0 +1,81 @@
+// Copyright 2025 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "DolphinQt/EditTimePlayedDialog.h"
+
+#include <chrono>
+#include <string>
+
+#include <QDialog>
+#include <QDialogButtonBox>
+#include <QFormLayout>
+#include <QLabel>
+#include <QSpinBox>
+#include <QString>
+#include <QWidget>
+
+#include "Core/TimePlayed.h"
+#include "Core/TitleDatabase.h"
+
+#include "UICommon/GameFile.h"
+
+EditTimePlayedDialog::EditTimePlayedDialog(QWidget* const parent, const UICommon::GameFile& game)
+    : QDialog(parent), m_game_id(game.GetGameID())
+{
+  using namespace std::chrono_literals;
+
+  // std::chrono::milliseconds is guaranteed to have a signed integer representation type with at
+  // least 45 bits (44 non-sign bits). (2^44)/(3'600'000 milliseconds/hour) ~= 4'886'718 hours.
+  // Setting MAX_EDITABLE_HOURS to 999'999 makes all <= 6 digit hour counts valid and corresponds to
+  // 114 years, which is more than anyone will actually ever play.
+  constexpr auto MAX_EDITABLE_HOURS = std::chrono::hours(999'999);
+  constexpr std::chrono::milliseconds MAX_EDITABLE_TIME =
+      MAX_EDITABLE_HOURS + std::chrono::minutes(59);
+
+  const std::chrono::milliseconds time_played =
+      std::min(MAX_EDITABLE_TIME, TimePlayedManager::GetInstance().GetTimePlayed(m_game_id));
+  const auto hours_played = time_played / 1h;
+  const auto minutes_played = (time_played % 1h) / 1min;
+
+  setWindowTitle(QStringLiteral("Edit Time Played"));
+
+  auto* const layout = new QFormLayout;
+
+  m_hours_edit = new QSpinBox;
+  m_hours_edit->setRange(0, MAX_EDITABLE_HOURS.count());
+  m_hours_edit->setValue(hours_played);
+
+  m_minutes_edit = new QSpinBox;
+  m_minutes_edit->setRange(0, 59);
+  m_minutes_edit->setValue(minutes_played);
+
+  auto* const button_box = new QDialogButtonBox(QDialogButtonBox::Cancel | QDialogButtonBox::Save);
+
+  const Core::TitleDatabase title_db;
+  layout->addRow(tr("Game:"), new QLabel(QString::fromStdString(game.GetName(title_db))));
+  layout->addRow(tr("ID:"), new QLabel(QString::fromStdString(m_game_id)));
+  layout->addRow(tr("Hours:"), m_hours_edit);
+  layout->addRow(tr("Minutes:"), m_minutes_edit);
+  layout->addWidget(button_box);
```

Comment:
```suggestion
  layout->addRow(button_box);
  
  layout->setFieldGrowthPolicy(QFormLayout::AllNonFixedFieldsGrow);
```
using `addRow` fixes the buttons being forced to the right side of the form layout.

`setFieldGrowthPolicy` makes the two `QSpinBox` the same width.

Reply:
Huh, guess this is a platform difference. On Windows these changes don't do anything; the buttons are on the right for every dialog (even when using `addRow`), and the fields already grow to the edge of the form and have the same width.

Since it doesn't hurt on Windows and apparently helps on Linux I'll go ahead and add them.

Reply:
before and after:
![image](https://github.com/user-attachments/assets/b159e04f-3ee9-4cae-9087-58a4483d5ac7)
![image](https://github.com/user-attachments/assets/8d22b9c5-b8e0-40e8-bada-ff020b065e45)

Reply:
Ah, I misunderstood what you said; I thought you were talking about the buttons being right-aligned rather than being restricted from the left half of the form. `AddRow` vs. `AddWidget` behaves that way on Windows too.